src/map/lua/lua_baseentity.cpp

```diff
@@ -8692,7 +8674,17 @@ inline int32 CLuaBaseEntity::hasSpellList(lua_State* L)
     DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
     DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_MOB && m_PBaseEntity->objtype != TYPE_PET);
 
-    lua_pushboolean(L, ((CMobEntity*)m_PBaseEntity)->SpellContainer->HasSpells());
+    lua_pushboolean(L, (static_cast<CMobEntity *>(m_PBaseEntity))->SpellContainer->HasSpells());
+
+    return 1;
+}
+
+inline int32 CLuaBaseEntity::hasPreventActionEffect(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_NPC);
```

Comment:
Sure, I guess it probably should.  I was just thinking of it in the context of the code I was writing, but I suppose this function could be useful for players as well.  I’ll change it to DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_NPC);

 

From: Tahir Akhlaq [mailto:notifications@github.com] 
Sent: Thursday, March 3, 2016 11:02 AM
To: DarkstarProject/darkstar <darkstar@noreply.github.com>
Cc: lenonk <lenon@sanctuary.org>
Subject: Re: [darkstar] BST stay command now works per retail. The amount healed may need to be (#2849)

 

In src/map/lua/lua_baseentity.cpp <https://github.com/DarkstarProject/darkstar/pull/2849#discussion_r54910746> :

> @@ -8692,7 +8674,17 @@ inline int32 CLuaBaseEntity::hasSpellList(lua_State* L)
>      DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
>      DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_MOB && m_PBaseEntity->objtype != TYPE_PET);
>  
> -    lua_pushboolean(L, ((CMobEntity*)m_PBaseEntity)->SpellContainer->HasSpells());
> +    lua_pushboolean(L, (static_cast<CMobEntity *>(m_PBaseEntity))->SpellContainer->HasSpells());
> +
> +    return 1;
> +}
> +
> +inline int32 CLuaBaseEntity::hasPreventActionEffect(lua_State* L)
> +{
> +    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
> +    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_NPC);

applies to players too though? (battle entity)

—
Reply to this email directly or view it on GitHub <https://github.com/DarkstarProject/darkstar/pull/2849/files#r54910746> .  <https://github.com/notifications/beacon/ABIIJdg4xzHohXL-cez5d6Z8Cluqow4fks5ppxQVgaJpZM4HnQDO.gif>